charts/common/templates/_drax.tpl

```diff
@@ -66,3 +66,29 @@ The Role this component serves in the overall application
 
 {{- ($values.drax).role | default ($.Values.drax).role -}}
 {{- end -}}
+
+
+{{/*
+The Accelleran Product set for the deployment
+*/}}
+{{- define "accelleran.common.drax.accelleranProduct" -}}
+{{- $ := get . "top" | required "The top context needs to be provided to common drax accelleranProduct" -}}
+{{- $values := get . "values" | default $.Values -}}
+{{- $accelleranProduct := $values.global.accelleranProduct -}}
+
+{{- $allowedProducts := list "P5G" "DRAX" }}
+
+{{- $validProduct := false -}}
+{{- range $allowedProducts -}}
+    {{- if eq $accelleranProduct . -}}
+        {{- $validProduct = true -}}
+        {{- break -}}
+    {{- end -}}
+{{- end -}}
+
+{{- if not $validProduct -}}
+    {{- fail (printf "Error: global.accelleranProduct must be one of [%s]. Current value: '%s'" (join ", " $allowedProducts) $accelleranProduct) -}}
+{{- end -}}
+
+{{- $accelleranProduct | default ($.Values.global).accelleranProduct -}}
```

Comment:
This default could be dangerous as no check would have been done on it (and it would be possible to mess this up if we are passing values different than `$.Values`). Instead we should provide the default at the top as part of the assignment to `$accelleranProduct`.

```suggestion
{{- $accelleranProduct -}}
```